storage/src/mmr/mem.rs

```diff
@@ -55,6 +55,7 @@ impl<H: CHasher> Mmr<H> {
 
     // Return an `Mmr` initialized with the given nodes and oldest remembered position.
     pub fn init(nodes: Vec<H::Digest>, oldest_remembered_pos: u64) -> Self {
+        assert!(PeakIterator::check_validity(nodes.len() as u64));
```

Comment:
If we assert here, can we recover?

Or is the idea that that the stateful one will handle this (and the memory one should never be invalid)?

Reply:
Yeah I'm leaving this to the caller (the stateful MMR implementation) to ensure that only a valid MMR should be instantiated here.

Reply:
this was buggy (caught by a test I forgot to run), just fixed